weldx/tests/test_time.py

```diff
@@ -0,0 +1,296 @@
+    if cls_type is str:
+        return values
+    return cls_type(values)
+
+
+def _initialize_date_time_quantity(timedelta, unit, time_ref):
+    quantity = Q_(timedelta, unit)
+    setattr(quantity, "time_ref", Timestamp(time_ref))
+    return quantity
+
+
+def _is_timedelta(cls_type):
+    # todo: Q_ must be checked for attribute
+    return cls_type in [Q_, TimedeltaIndex, Timedelta, np.timedelta64] or (
+        cls_type is Time and not Time.is_absolute
+    )
+
+
+def _is_datetime(cls_type):
+    return not _is_timedelta(cls_type)
+
+
+class TestTime:
+    """Test the time class."""
+
+    # test_init helper functions -------------------------------------------------------
+
+    @staticmethod
+    def _transform_array(data, is_array, is_scalar):
+        if not is_array:
+            return data[0]
+        if is_scalar:
+            return [data[0]]
+        return data
+
+    @classmethod
+    def _create_init_input_type(
+        cls, input_type, delta_val, abs_val, is_timedelta, is_array, is_scalar
+    ):
+        """Create an instance of the desired input type for the `__init__` test."""
+        val = delta_val if is_timedelta else abs_val
+        if not is_timedelta and input_type is Q_:
+            val = [v - 1 for v in delta_val]
+        val = cls._transform_array(val, is_array=is_array, is_scalar=is_scalar)
+
+        # create the time input ---------------------------------
+        if is_timedelta:
+            return _initialize_delta_type(input_type, val, "s")
+        elif input_type is not Q_:
+            return _initialize_datetime_type(input_type, val)
+        return _initialize_date_time_quantity(val, "s", abs_val[0])
+
+    @classmethod
+    def _get_init_exp_values(
+        cls,
+        is_timedelta,
+        time_ref,
+        data_was_scalar,
+        delta_val,
+        abs_val,
+    ):
+        """Get the expected result values for the `__init__` test."""
+        exp_is_absolute = time_ref is not None or not is_timedelta
+
+        # expected reference time
+        exp_time_ref = None
+        if exp_is_absolute:
+            exp_time_ref = Timestamp(time_ref if time_ref is not None else abs_val[0])
+
+        # expected time delta values
+        val = delta_val
+        if exp_is_absolute:
+            offset = 0
+            if not is_timedelta:
+                if time_ref is not None:
+                    offset = Timestamp(abs_val[0]) - Timestamp(time_ref)
+                    offset = offset.total_seconds()
+                offset -= delta_val[0]
+
+            val = [v + offset for v in delta_val]
+
+        val = val[0] if data_was_scalar else val
+        exp_timedelta = (
+            Timedelta(val, "s") if data_was_scalar else TimedeltaIndex(val, "s")
+        )
+
+        # expected datetime
+        exp_datetime = None
+        if exp_is_absolute:
+            time_ref = Timestamp(time_ref if time_ref is not None else abs_val[0])
+            exp_datetime = time_ref + exp_timedelta
```

Comment:
This function describes the behavior that we expect from our `Time` class for the different input types. 
If `exp_datetime` is `None` we expect a `TypeError` 😉